Allow cluster-scoped instance CRDs via schema scope

antcybersec opened the pull request:
Adds spec.schema.scope (Namespaced|Cluster, immutable, default Namespaced) to ResourceGraphDefinition. CRD synthesis now sets scope accordingly, and instance reconciler handles cluster-scoped instances without requiring a namespace. Includes unit tests and docs note.

<img width="736" height="553" alt="image" src="https://github.com/user-attachments/assets/894d2201-7e21-4509-bcf4-d08b0cc7c46f" />


Fixes: https://github.com/kubernetes-sigs/kro/issues/806

a-hilaly reviewed (commented):
Thank you @antcybersec!  i do have one comment on the implementation.  however I do think that this feature deserves a KREP-XXX document that shares benefits, tradeoffs and (maybe) potential impact.

antcybersec commented:
hey any update on the PR @a-hilaly @tjamet

jakobmoellerdev commented:
/close

closing for now, feel free to reopen if you want to work on it again

antcybersec commented:
Hi @jakobmoellerdev @a-hilaly, I've addressed the feedback:

1.  **Created KREP document**: Added design proposal at docs/design/proposals/cluster-scoped-instances.md documenting:
   - Problem statement and use cases
   - Design details and API changes
   - Benefits, tradeoffs, and potential impact
   - Testing strategy
   - Backward compatibility considerations

2.  **Rebased**: Branch is now rebased on latest main and all conflicts resolved

3.  **Fixed issues**: 
   - Instance node now correctly sets Namespaced field based on scope
   - Status updates handle cluster-scoped instances
   - All tests passing

The PR is ready for review. The KREP document provides the design rationale and impact analysis you requested.

antcybersec commented:
> /close
> 
> closing for now, feel free to reopen if you want to work on it again

hey @jakobmoellerdev could you please  reopen the PR?

jakobmoellerdev commented:
no problem. since you force pushed you would have to open a fresh PR. I suggest you first create a KREP PR only so it will be easier to review and merge
